supporting.py

```diff
@@ -391,8 +391,14 @@ def implement_k_overlapped_alg(inst, primary_routes, extended_routes, capacity,
         else:
             workload[j] = max(0, primary_demands[j] - excess[j - 1])
 
-        excess[j] = min(capacity * np.ceil(float(workload[j]) / capacity) - workload[j], overlap_demands[j])
+        if workload[j] == 0:
+            excess[j] = excess[j-1] - primary_demands[j]
+            demand_filled[j] = 0
+        else:
+            excess[j] = min(capacity * np.ceil(float(workload[j]) / capacity) - workload[j], overlap_demands[j])
+            demand_filled[j] = workload[j] + excess[j]
         remaining_surplus = excess[j]
+        demand_filled = [demand_filled[j] for j in range(len(primary_routes))] # aligned with the datatype in create_full_trips()
```

Comment:
Is line 401 needed? In line 383, we set demand_filled to a numpy array of zeros. Then in lines 396 and 399, we overwrite the zero entry with the true value.